src/attribute_type.rs

```diff
@@ -1905,146 +1905,153 @@ fn parse_fmtp(to_parse: &str) -> Result<SdpAttribute, SdpParserInternalError> {
         unknown_tokens: Vec::new(),
     };
 
-    if parameter_token.contains('=') {
-        let parameter_tokens: Vec<&str> = parameter_token.split(';').collect();
-
-        for parameter_token in parameter_tokens.iter() {
-            let name_value_pair: Vec<&str> = parameter_token.splitn(2, '=').collect();
-
-            if name_value_pair.len() != 2 {
-                return Err(SdpParserInternalError::Generic(
-                    "A fmtp parameter must be either a telephone event, a parameter list or
+    for parameter_token in tokens[1..].to_vec() {
```

Comment:
From a vec to a slice back to a vec again doesn't seem very efficient.

I think converting this from the initial collect() call to using the next() call on the iterator would be better.

Reply:
Fixed. I changed the to_vec() to iter() as to avoid the use of next().